syftr/ray/submit.py

```diff
@@ -26,9 +27,14 @@ def get_client() -> JobSubmissionClient:
 
 
 def _get_metadata(study_config: StudyConfig) -> Dict[str, Any]:
-    repo = git.Repo(cfg.paths.root_dir)
-    sha = repo.head.commit.hexsha
-    short_sha = repo.git.rev_parse(sha, short=True)
+    try:
+        repo = git.Repo(cfg.paths.root_dir)
+        sha = repo.head.commit.hexsha
+        short_sha = repo.git.rev_parse(sha, short=True)
+    except git.exc.InvalidGitRepositoryError:
+        # We are not in a git repo, syftr used as a library, so generating a random sha.
+        short_sha = secrets.token_hex(nbytes=8)
+        sha = "library_" + short_sha
```

Comment:
Actually, it's better to just not have a git_sha or git_short_sha if we aren't in a git repo.

Reply:
It still requires some random string to distinguish between submissions.

Reply:
I can rename dict keys if necessary.

Reply:
oh, haven't noticed datetime is there. updated.